### PR TITLE
fix/race condition hard reset

### PR DESCRIPTION
**Pull request by 0xArdi**
## Fixes

This PR showcases the hard reset race condition seen in #1583.

For hard reset to work, the following steps must be strictly followed:
1. Tendermint server receives /hard_reset request from the agent.
2. Tendermint server stops tendermint.
3. Tendermint server performs unsafe-reset-all.
4. Tendermint server restarts tendermint.
5. Agent receives success response, and resets local blockchain. 
6. Tendermint sends a handshake info request to the agent.

In this failure step 6 happens before step 5. 

**The fix will be applied in a separate PR.**
  
## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [x] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [x] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have locally run services that could be impacted and they do not present failures derived from my changes

### CODE REVIEW

File: packages/valory/agents/register_reset/tests/helpers/slow_tendermint_server/Dockerfile
Comment (0xArdi) [inline packages/valory/agents/register_reset/tests/helpers/slow_tendermint_server/Dockerfile:1]: copied over from `deployments/Dockerfiles/tendermint`

File: packages/valory/agents/register_reset/tests/helpers/slow_tendermint_server/tendermint.py
Comment (0xArdi) [inline packages/valory/agents/register_reset/tests/helpers/slow_tendermint_server/tendermint.py:1]: deployments/Dockerfiles/tendermint

File: packages/valory/agents/register_reset/tests/helpers/slow_tendermint_server/wrapper.sh
Comment (0xArdi) [inline packages/valory/agents/register_reset/tests/helpers/slow_tendermint_server/wrapper.sh:1]: copied from deployments/Dockerfiles/tendermint

File: packages/valory/agents/register_reset/tests/test_hard_reset_race_condition.py
Comment (0xArdi) [inline packages/valory/agents/register_reset/tests/test_hard_reset_race_condition.py:40]: import the custom fixture to run with the new image

**Review by dvilelaf — COMMENTED** [review]
LGTM, once the top PR is merged this should pass.

File: .github/workflows/main_workflow.yml
Comment (0xArdi) [reply .github/workflows/main_workflow.yml:405]: This new image is required to showcase the problem.

File: packages/valory/agents/register_reset/tests/helpers/conftest.py
Comment (0xArdi) [reply packages/valory/agents/register_reset/tests/helpers/conftest.py:1]: I don't see a way to use the new image without having this. We need the new image to show the problem.

File: .github/workflows/main_workflow.yml
Comment (0xArdi) [reply .github/workflows/main_workflow.yml:405]: We cannot use a unit test as the actual error is thrown on the tendermint side. 
This new image simply "lags" in the hard reset response to the agent, leading to the scenario laid out in the description of the PR.

Comment (DavidMinarsch) [reply .github/workflows/main_workflow.yml:405]: Ok I see - so it is not used in production

File: packages/valory/agents/register_reset/tests/helpers/conftest.py
Comment (DavidMinarsch) [reply packages/valory/agents/register_reset/tests/helpers/conftest.py:1]: Ok - can you then add a lot of description in the `__init__` file of this folder so it is very clear why we have this?

File: .github/workflows/main_workflow.yml
Comment (0xArdi) [reply .github/workflows/main_workflow.yml:405]: Correct, only used for this specific test. I have added disclaimers in the docstring to not use this anywhere else, as it might lead to unwanted behaviour.

**Review by DavidMinarsch — APPROVED** [review]

**Pull request merged**
